### PR TITLE
Add the possibility to change the addressRange maxLen with options.

### PR DESCRIPTION
**Pull request by Hurtsich**
Hello, we have a national service provider that use a custom protocol based on SMPP.
We need to have the possibility to set the max length for the address range.

I made the change using the option pattern (like the change on the connector to inject the address range)
This way we open the lib but do not break the public interface.

### CODE REVIEW

**Comment by laduchesneau** [comment]
Hi, I haven't looked over the code yet, but if I may propose another solution to this problem....lets make all Address fields exportalable.  We don't need getters and setters. Default TON/NPI with be both be 0 anyway. 

I would delete it all and just add a validator function

``` golang
// Address smpp address of src and dst.
type Address struct {
	Ton     byte
	Npi     byte
	Address string
}

// isValid verifies the Address, return a error if not valid
func (c *Address) isValid() error{
	if len(c.Address) > data.SM_ADDR_LEN {
		return fmt.Errorf("address len exceed limit. (%d > %d)", len(c.Address), data.SM_ADDR_LEN)
	}
	return nil
}
```

We would still keeps the marshal, unmarshal and string functions

This would be a breaking change, but a good one. If this solution is accepted, I would also recommend the same change for AddressRange.

**Comment by linxGnu** [comment]
@Hurtsich 
Simplest solutions:
- Solution 1: Exposing `AddressRange` field
- Solution 2: implement function like this:
```go
func (c * AddressRange) ForceSetAddressRange(addr string){
     c.addressRange = addr
}
```

Just it if you want to have custom addrRange without validation.

**Comment by linxGnu** [comment]
@laduchesneau 
yeah, I understand your concern. I think best practice here is: only "open" if needed/on user request.
In this way, we balanced between maintenance effort and user need.

**Comment by Hurtsich** [comment]
@linxGnu If you're OK with breaking the public interface, i think just exposing the fields on AddressRange is fine.

**Comment by Hurtsich** [comment]
Ping, I updated the PR with the changes that we discussed. I'll let you merge if it's okay :)

File: pdu/AddressRange.go [hunk 1]
```diff
@@ -1,92 +1,27 @@
...
-
-// NewAddressRangeWithTonNpi create new AddressRange with ton, npi.
-func NewAddressRangeWithTonNpi(ton, npi byte) AddressRange {
-	return AddressRange{ton: ton, npi: npi}
-}
-
-// NewAddressRangeWithTonNpiAddr returns new address with ton, npi, addr string.
-func NewAddressRangeWithTonNpiAddr(ton, npi byte, addr string) (a AddressRange, err error) {
-	a = NewAddressRangeWithTonNpi(ton, npi)
-	err = a.SetAddressRange(addr)
```
Comment (linxGnu) [inline pdu/AddressRange.go:36]: Could you please keep these "Constructor" to prevent breaking changes?

**Review by linxGnu — APPROVED** [review]
LGTM. Thank you very much.

**Pull request merged**